barretenberg/cpp/src/barretenberg/vm/avm_trace/constants.hpp

```diff
@@ -67,8 +67,10 @@ inline const uint32_t PCPI_NEW_NULLIFIERS_OFFSET =
 inline const uint32_t PCPI_NEW_L2_TO_L1_MSGS_OFFSET =
     PCPI_NEW_NULLIFIERS_OFFSET + (MAX_NEW_NULLIFIERS_PER_CALL * NULLIFIER_LENGTH);
 
-inline const uint32_t PCPI_NEW_UNENCRYPTED_LOGS_OFFSET /*2 item gap*/ =
-    PCPI_NEW_L2_TO_L1_MSGS_OFFSET + (MAX_NEW_L2_TO_L1_MSGS_PER_CALL * L2_TO_L1_MESSAGE_LENGTH) + 2;
+inline const uint32_t PCPI_START_SIDE_EFFECT_COUNTER_OFFSET =
+    PCPI_NEW_L2_TO_L1_MSGS_OFFSET + (MAX_NEW_L2_TO_L1_MSGS_PER_CALL * L2_TO_L1_MESSAGE_LENGTH);
+
+inline const uint32_t PCPI_NEW_UNENCRYPTED_LOGS_OFFSET = PCPI_START_SIDE_EFFECT_COUNTER_OFFSET + /*1 item gap*/ 1;
```

Comment:
This offset thing is so bad :sweat_smile: we need to figure out a proper way to do this

Reply:
agreed